### PR TITLE
[GOBBLIN-785] remove wrapper isPartition function, use table.isPartitioned instead

### PR DESCRIPTION
**Pull request by jhsenjaliya**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-785

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
some places use HiveUtils.isPartitioned() which is un-necessary wrapper function, and some places uses direct HiveTable.isPartitioned() the later should be used consistently.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: no change or addition in functionality.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java [hunk 1]
```diff
@@ -163,10 +163,4 @@ private static Configuration getHadoopConfiguration() {
     return conf;
   }
 
-  /**
-   * @return true if {@link Table} is partitioned.
-   */
-  public static boolean isPartitioned(Table table) {
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java]: Please leave the method and mark as deprecated. It is possible other products are relying on this method as a library.

Comment (jhsenjaliya) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java]: Sure, added method back with deprecation note. Thanks

**Comment by ibuenros** [comment]
@jhsenjaliya I see a test failure that I don't see in other PRs. Could you check?

**Comment by jhsenjaliya** [comment]
@ibuenros , fixed it, pls take a look, Thanks

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java (hunk 1 above)
Comment (ibuenros) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java]: I don't see the method added back, did you push the change?

**Comment by jhsenjaliya** [comment]
@ibuenros , ya that commit was missed, added back, Thanks

**Comment by shirshanka** [comment]
Looks like all review comments have been addressed. 
@ibuenros : can you re-check? 
LGTM!

**Comment by arjun4084346** [comment]
LGTM

**Comment by jhsenjaliya** [comment]
@sv2000 , pls take a look and merge this. Thanks

**Pull request closed without merging**